pre_build/cccp_index_pre_build_check.py

```diff
@@ -116,48 +116,44 @@ def main(indexdlocation):
     tempfiles = []
     for project in get_projects_from_index(indexdlocation):
         try:
-            print("Processing project with details: %s", str(project))
+            print("Processing project with details: %s" % str(project))
             try:
                 env = Environment(loader=FileSystemLoader(
                     './'), trim_blocks=True, lstrip_blocks=True)
                 template = env.get_template(jjb_defaults_file)
                 job_details = template.render(project[0]['project'])
             except Exception as e:
-                print("Error template is not updated: %s", str(e))
+                print("Error template is not updated: %s" % str(e))
 
             try:
-                t = tempfile.mkdtemp()
-                generated_filename = os.path.join(
-                    t,
-                    'cccp_GENERATED.yaml'
-                )
+                pre_text = ''.join(random.sample(
+                    string.lowercase + string.digits, 10))
+                generated_filename = '_'.join(pre_text,
+                                              'cccp_GENERATED.yaml'
+                                              )
```

Comment:
so `randomString_cccp_GENERATED.yml` is going to be inside workspace directory and not in temporary directory?

Reply:
yup, and every run cleans the workspace.